pkg/k8shandler/status_test.go

```diff
@@ -366,3 +369,86 @@ func populateNodes(clusterName string, objs []runtime.Object, client client.Clie
 	}
 	return nodes
 }
+
+func TestUnschedulableNodesConditions(t *testing.T) {
+
+	testCluster := &loggingv1.Elasticsearch{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "test-namespace",
+			Name:      "elasticsearch",
+		},
+		Status: loggingv1.ElasticsearchStatus{
+			Nodes: []loggingv1.ElasticsearchNodeStatus{
+				{
+					DeploymentName: "elasticsearch-cdm-lvmn62il-1",
+				},
+			},
+		},
+	}
+
+	nowTime, _ := time.Parse("RFC3339", "2020-10-12T20:40:18Z")
+	transitionTime := metav1.NewTime(nowTime)
+
+	mockedPodList := &v1.PodList{
+		Items: []v1.Pod{
+			{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Pod",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"cluster-name":   "elasticsearch",
+						"component":      "elasticsearch",
+						"es-node-client": "true",
+						"es-node-data":   "true",
+						"es-node-master": "true",
+						"node-name":      "elasticsearch-cdm-lvmn62il-1",
+					},
+					Namespace: "test-namespace",
+				},
+				Status: v1.PodStatus{
+					Conditions: []v1.PodCondition{
+						{
+							Type:               v1.PodScheduled,
+							Status:             v1.ConditionFalse,
+							Reason:             v1.PodReasonUnschedulable,
+							Message:            "0/6 nodes are available: 6 Insufficient memory.",
+							LastTransitionTime: transitionTime,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	mockClient := fake.NewFakeClient(mockedPodList)
+	er := &ElasticsearchRequest{client: mockClient, cluster: testCluster}
+
+	// call updatePodNodeConditions and verify we have the correct node conditions
+
+	status := testCluster.Status.DeepCopy()
+
+	err := er.updatePodNodeConditions(status, false)
+
+	if err != nil {
+		t.Errorf("Received error while testing updating pod node conditions: %v", err)
+	}
+
+	testCluster.Status.Nodes[0].Conditions = []loggingv1.ClusterCondition{
+		{
+			Type:               loggingv1.Unschedulable,
+			Reason:             "Unschedulable",
+			Status:             v1.ConditionTrue,
+			Message:            "0/6 nodes are available: 6 Insufficient memory.",
+			LastTransitionTime: transitionTime,
+		},
+	}
+
+	t.Logf("expected: %v", testCluster.Status.Nodes)
+	t.Logf("actual: %v", status.Nodes)
+
+	if !reflect.DeepEqual(testCluster.Status.Nodes, status.Nodes) {
+		t.Errorf("Expected cluster node statuses to be same")
+	}
```

Comment:
I suggest to replace this with `cmp.Diff` instead of logging each one individually and then comparing with `reflect.DeepEqual`